stream/src/main/scala/org/apache/pekko/stream/scaladsl/Source.scala

```diff
@@ -930,6 +931,47 @@ object Source {
   def queue[T](bufferSize: Int): Source[T, BoundedSourceQueue[T]] =
     Source.fromGraph(new BoundedSourceQueueStage[T](bufferSize))
 
+  /**
+   * Creates a Source that will immediately execute the provided function `f` with a [[BoundedSourceQueue]] when materialized.
+   * This allows defining element production logic at Source creation time.
+   *
+   * The function `f` can push elements to the stream using the provided queue. The queue behaves the same as in [[Source.queue]]:
+   * <br>
+   * - Elements are emitted when there is downstream demand, buffered otherwise
+   * <br>
+   * - Elements are dropped if the buffer is full
+   * <br>
+   * - Buffered elements are discarded if downstream terminates
+   *
+   * If the function `f` throws an exception, the queue will be failed and the exception will be propagated to the stream.
+   *
+   * Example usage:
+   * {{{
+   * Source.create[Int](10) { queue =>
+   *   // This code is executed when the source is materialized
+   *   queue.offer(1)
+   *   queue.offer(2)
+   *   queue.offer(3)
+   *   queue.complete()
+   * }
```

Comment:
I think this is much nicer than the old one for some integration cases.

Reply:
It would be better if change the parameter name `f` of the method.

Reply:
To which？

Reply:
Name it emitter?